src/System.Threading.Timer/tests/TimerChangeTests.cs

```diff
@@ -0,0 +1,120 @@
+            {
+                t.Change(new TimeSpan(1) /* not relevant */, TimeSpan.FromMilliseconds(-2));
+            }));
+        }
+    }
+
+    [Fact]
+    public static void Timer_Change_NegativeInt_DueTime_Throws()
+    {
+        using (var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1))
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(new Action(() =>
+            {
+                t.Change(-2, 1 /* not relevant */);
+            }));
+        }
+    }
+
+    [Fact]
+    public static void Timer_Change_NegativeInt_Period_Throws()
+    {
+        using (var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1))
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(new Action(() =>
+            {
+                t.Change(1 /* not relevant */, -2);
+            }));
+        }
+    }
+    [Fact]
+    public static void Timer_Change_TooLongTimeSpan_DueTime_Throws()
+    {
+        using (var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1))
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(new Action(() =>
+            {
+                t.Change(TimeSpan.FromMilliseconds((long)0xFFFFFFFF), new TimeSpan(1) /* not relevant */);
+            }));
+        }
+    }
+
+    [Fact]
+    public static void Timer_Change_TooLongTimeSpan_Period_Throws()
+    {
+        using (var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1))
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(new Action(() =>
+            {
+                t.Change(new TimeSpan(1) /* not relevant */, TimeSpan.FromMilliseconds((long)0xFFFFFFFF));
+            }));
+        }
+    }
+
+    [Fact]
+    public static void Timer_Change_TimeSpan_AfterDispose_Throws()
+    {
+        var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1);
+        t.Dispose();
+        Assert.Throws<ObjectDisposedException>(new Action(() =>
+        {
+            t.Change(TimeSpan.FromMilliseconds(1), TimeSpan.FromMilliseconds(1));
+        }));
+    }
+
+    [Fact]
+    public static void Timer_Change_Int_AfterDispose_Throws()
+    {
+        var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1);
+        t.Dispose();
+        Assert.Throws<ObjectDisposedException>(new Action(() =>
+        {
+            t.Change(1, 1);
+        }));
+    }
+
+    [Fact]
+    public static void Timer_Change_BeforeDueTime_ChangesWhenTimerWillFire()
+    {
+        AutoResetEvent are = new AutoResetEvent(false);
+
+        using (var t = new Timer(new TimerCallback((object s) =>
+        {
+            are.Set();
+        }), null, TimeSpan.FromMilliseconds(500), TimeSpan.FromMilliseconds(50)))
+        {
+            t.Change(TimeSpan.FromSeconds(1), TimeSpan.FromMilliseconds(-1));
+            Assert.False(are.WaitOne(TimeSpan.FromMilliseconds(600)), "Should not have had a timer event since we changed the DueTime");
```

Comment:
These values will make the test take a while and also seem a bit prone to random fluctations in the CI system.  What if you instead set the timeout to something extremely large, verified that the event isn't set, then change it to something extremely small, and verify that it fires?